File: fix_wait.py
import sys

file_path = 'e:/AUTOFLOW/AUTOMATION_FLOW/components/flows/tabs/FlowAnalyticsTab.tsx'
with open(file_path, 'r', encoding='utf-8') as f:
    content = f.read()

# 1. Remove shifting logic
target_shift = '''        // Loop through steps in order and push 'wait' counts forward
        // Repeat up to 3 times to handle chained wait nodes (Wait -> Wait -> Action)
        for (let i = 0; i < 3; i++) {
            sortedSteps.forEach(s => {
                if (s.type === 'wait') {
                    const count = shiftedWaiting.get(s.id) || 0;
                    if (count > 0 && s.nextStepId) {
                        const currentTargetCount = shiftedWaiting.get(s.nextStepId) || 0;
                        shiftedWaiting.set(s.nextStepId, currentTargetCount + count);
                        shiftedWaiting.set(s.id, 0); // Clear from wait step
                    }
                }
            });
        }'''
content = content.replace(target_shift, '')

# 2. Allow effectiveStepId shifting logic to NOT shift wait
target_fetch = '''        // [SHIFTING FIX] If we are looking for 'waiting' users for a specific step,
        // we must also include users who are at any 'wait' step that leads to this step.
        let effectiveStepId = stepId;
        if (stepId && (!status || status === 'waiting')) {
            const sources = [stepId];
            const findWaitSources = (targetId: string) => {
                currentFlow.steps.forEach(s => {
                    if (s.type === 'wait' && s.nextStepId === targetId && !sources.includes(s.id)) {
                        sources.push(s.id);
                        findWaitSources(s.id);
                    }
                });
            };
            findWaitSources(stepId);
            effectiveStepId = sources.join(',');
        }'''
content = content.replace(target_fetch, '        let effectiveStepId = stepId;')

# 3. Remove item.type !== 'wait' from click handler
target_click = '''                                                    onClick={() => item.type !== 'wait' && handleStepClick(item.id, item.type)}'''
content = content.replace(target_click, '''                                                    onClick={() => handleStepClick(item.id, item.type)}''')

# 4. Remove special card styling for wait step
target_style = '''\'''
content = content.replace(target_style, '''\''')

target_style2 = '''(item.type === 'wait' ? 'border-slate-200 dark:border-slate-700/60' : 'border-slate-100 dark:border-slate-800/60 hover:shadow-lg hover:border-slate-200 dark:border-slate-700/60')'''
content = content.replace(target_style2, '''(item.type === 'wait' ? 'border-slate-300 dark:border-slate-700 hover:border-slate-400' : 'border-slate-100 dark:border-slate-800/60 hover:shadow-lg hover:border-slate-200 dark:border-slate-700/60')''')

# 5. Remove 'Ch?: {item.waiting}' badge
target_badge = '''                                                                    {item.waiting > 0 && (
                                                                        <span className="flex items-center gap-1 text-[8px] md:text-[9px] font-bold text-amber-600 bg-amber-50 px-1.5 py-0.5 rounded-md animate-pulse">
                                                                            <div className="w-1 h-1 md:w-1.5 md:h-1.5 rounded-full bg-amber-600"></div>
                                                                            Ch?: {item.waiting}
                                                                        </span>
                                                                    )}'''
content = content.replace(target_badge, '')

# 6. Show pills for wait step
target_pills = ''') : item.type !== 'wait' ? ('''
content = content.replace(target_pills, ''') : (''')

with open(file_path, 'w', encoding='utf-8') as f:
    f.write(content)

print('Success')

File: fix_wait.js
const fs = require('fs');
const path = 'e:/AUTOFLOW/AUTOMATION_FLOW/components/flows/tabs/FlowAnalyticsTab.tsx';
let content = fs.readFileSync(path, 'utf8');

content = content.replace(
    /[\s]*\/\/ Loop through steps in order and push 'wait' counts forward[\s\S]*?\/\/ Clear from wait step[\s\S]*?\}[\s]*\}\);[\s]*\}[\s]*/,
    '\n\n        // Removed shift loop\n\n'
);

content = content.replace(
    /[\s]*\/\/ \[SHIFTING FIX\] If we are looking for 'waiting' users for a specific step,[\s\S]*?effectiveStepId = sources\.join\(\',\',\);[\s]*\}[\s]*/,
    '\n        // Removed shifting fix\n        let effectiveStepId = stepId;\n'
);

content = content.replace(
    /onClick=\{\(\) => item\.type !== 'wait' && handleStepClick\(item\.id, item\.type\)\}/g,
    'onClick={() => handleStepClick(item.id, item.type)}'
);

content = content.replace(
    /\$\{item\.type === 'wait' \? 'p-2 md:p-3 border-dashed bg-slate-100\/50 opacity-90' : 'bg-white dark:bg-slate-900 p-3 md:p-4 cursor-pointer'\}/g,
    '\'
);

content = content.replace(
    /\(item\.type === 'wait' \? 'border-slate-200 dark:border-slate-700\/60' : 'border-slate-100 dark:border-slate-800\/60 hover:shadow-lg hover:border-slate-200 dark:border-slate-700\/60'\)/g,
    '(item.type === \'wait\' ? \'border-slate-300 dark:border-slate-700 hover:border-slate-400\' : \'border-slate-100 dark:border-slate-800/60 hover:shadow-lg hover:border-slate-200 dark:border-slate-700/60\')'
);

content = content.replace(
    /[\s]*\{item\.waiting > 0 && \([\s]*<span className=\"flex items-center gap-1 text-\[8px\] md:text-\[9px\] font-bold text-amber-600 bg-amber-50 px-1\.5 py-0\.5 rounded-md animate-pulse\">[\s]*<div className=\"w-1 h-1 md:w-1\.5 md:h-1\.5 rounded-full bg-amber-600\"><\/div>[\s]*Ch?: \{item\.waiting\}[\s]*<\/span>[\s]*\)\}/,
    ''
);

content = content.replace(
    /\) : item\.type !== 'wait' \? \(/g,
    ') : ('
);

fs.writeFileSync(path, content, 'utf8');
console.log('Done');
